### PR TITLE
Make scripts/variables more "canonical" and unified across librosa/scipy/torch versions, Fix synthesis window in torch version, add functionality of choosing window type in scipy version, also make use of the Torch-built-in ISTFT function in the similar way as we use the STFT one

### PR DESCRIPTION
**Pull request by tz579**
1. Make scripts/variables more "canonical" in torch version, and make it more unified with librosa/scipy versions
2. Fix synthesis window in torch version
3. Add functionality of choosing window type in scipy version

Updated on Apr. 08, 2019: 
1. fix errors of test_torch.py that prevent test.py from running correctly
2. further remove some unnecessary modifications of test_torch.py done previously
3. better handling of input/output for optional batch dim, numpy/torch.tensor array types, cpu/cuda devices for torch.tensor, etc.

Update on Apr. 18, 2022:
1. refactored to make use of the built-in ISTFT function in the similar way as we use STFT one.
2. other refinements.

### CODE REVIEW

**Comment by faroit** [comment]
oops, I somehow missed this PR earlier. Thanks for this, I will have a look later

**Comment by faroit** [comment]
@tz579 maybe now that torchaudio (and torch) has a builtin ISTFT, it would actually be nice to replace that and update the requirements

**Comment by tz579** [comment]
> @tz579 maybe now that torchaudio (and torch) has a builtin ISTFT, it would actually be nice to replace that and update the requirements

Thanks for that and sorry not to have replied you timely.

Now it has been further refactored to make use of the built-in ISTFT function in the similar way as we use STFT one, together with some other refinements.

Best,

**Comment by faroit** [comment]
@tz579 thats great! i will have a look in the next days!

**Comment by tz579** [comment]
> 

Thanks a lot!

**Comment by tz579** [comment]
> @tz579 thats great! i will have a look in the next days!

Thanks a lot!
